### PR TITLE
fix: pubsub auth

### PR DESCRIPTION
**Pull request by dermasmid**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

While trying to test risingwave with gcp pubsub, i was not able to get it to connect to GCP, failing with a error: `This is dummy token source provider. you can use 'google_cloud_default' crate` .

looking here https://github.com/yoshidan/google-cloud-rust/tree/main/pubsub it seems like we need to add `with_auth`, so i did, its possible that its a recent change on their end

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [x] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/source/google_pubsub/enumerator/client.rs [hunk 1]
```diff
@@ -51,7 +51,8 @@ impl SplitEnumerator for PubsubSplitEnumerator {
         properties.initialize_env();
 
         // Validate config
-        let client = Client::new(Default::default())
+        let config = ClientConfig::default().with_auth().await.unwrap();
```
Comment (xxchan) [inline src/connector/src/source/google_pubsub/enumerator/client.rs:54]:
```suggestion
        let config = ClientConfig::default().with_auth().await?;
```

File: src/connector/src/source/google_pubsub/source/reader.rs [hunk 2]
```diff
@@ -126,7 +126,8 @@ impl SplitReader for PubsubSplitReader {
         // Set environment variables consumed by `google_cloud_pubsub`
         properties.initialize_env();
 
-        let client = Client::new(Default::default())
+        let config = ClientConfig::default().with_auth().await.unwrap();
```
Comment (xxchan) [inline src/connector/src/source/google_pubsub/source/reader.rs:129]:
```suggestion
        let config = ClientConfig::default().with_auth()?;
```

**Review by xxchan — COMMENTED** [review]
Otherwise LGTM. Thanks!

BTW, did you tested that the problem is fixed?

**Review by yufansong — COMMENTED** [review]
Also LGTM. Please also add the test result in description, thanks.

**Comment by tabVersion** [comment]
Let me commit to their branch to fix this

**Comment by tabVersion** [comment]
resolved in https://github.com/risingwavelabs/risingwave/pull/14357

**Comment by tabVersion** [comment]
checkout to https://github.com/risingwavelabs/risingwave/pull/14360

**Pull request closed without merging**